Reduce font-size for titles in PR list

donokuda opened the pull request:
The font size for titles in the pull request list were large, which tends to draw the eye away from the main area of Visual Studio. This PR reduces the font size back to the default size and adds text trimming for long PR titles.

**Before**

![image](https://cloud.githubusercontent.com/assets/1174461/17570675/40b06cc2-5f02-11e6-9f32-b9bff1a78fa8.png)

**After**

![image](https://cloud.githubusercontent.com/assets/1174461/17573230/e5d0ebf4-5f0d-11e6-9e67-cc6677a13053.png)

donokuda commented on src/GitHub.VisualStudio.UI/Styles/ActionLinkButton.xaml:
```diff
@@ -10,7 +10,7 @@
       <Setter.Value>
         <ControlTemplate TargetType="Button">
-          <TextBlock>
+          <TextBlock TextTrimming="CharacterEllipsis">
```
Should I be aware of instances where we don't want this to be set? Is there a better way to approach this?

shana replied:
The better way is to have it as a templated setter (like the `Background`, `Focusable` and `Foreground` properties above). The problem there is that the `Button` object doesn't have a `TextTrimming` property (because buttons can have non-text content), so we need to switch from using a button to using a `GitHubActionLink` and add a custom DependencyProperty to it that we can then use to set the trimming.

shana commented:
#### I added a touch of SemiBold to the title:
![font-size-trimming-weight](https://cloud.githubusercontent.com/assets/310137/17585583/04807b00-5fbd-11e6-8236-9c61be5b0553.gif)

#### But then I removed it because it probably doesn't need it, so this is how it looks now:
![font-size-trimming-weight-normal](https://cloud.githubusercontent.com/assets/310137/17585712/7bcadeee-5fbd-11e6-9c00-60642b5d6ac2.gif)

grokys commented on src/GitHub.UI/Controls/Buttons/GitHubActionLink.cs:
```diff
@@ -8,12 +8,21 @@ public partial class GitHubActionLink : Button
             "HasDropDown", typeof(bool), typeof(GitHubActionLink));
 
+        public static readonly DependencyProperty TextTrimmingProperty = DependencyProperty.Register(
+            "TextTrimming", typeof(TextTrimming), typeof(GitHubActionLink));
```
Small thing, but you can register this property by adding an owner to the `TextBlock.TextTrimmingProperty` like so:

```
public static readonly DependencyProperty TextTrimmingProperty =
    TextBlock.TextTrimmingProperty.AddOwner(typeof(GitHubActionLink));
```

It probably won't make much difference to how the code runs but it tells the reader (at least it tells me) that this is an existing property that is being used, and that I can look up `TextBlock.TextTrimming` for more info.

shana replied:
Good point!

donokuda commented:
@shana @grokys 😍 Thanks for the insights around my question! This implementation is much, much cleaner.

> I added a touch of SemiBold to the title
But then I removed it because it probably doesn't need it, so this is how it looks now

Yeah, it's hard to notice that it's bolded slightly at that font size but it's fine without it being bold.

Here's where we're at after merging master into the branch:

![image](https://cloud.githubusercontent.com/assets/1174461/17599755/73043e5c-5fb5-11e6-8970-99ae74ccc256.png)

I think this is good to go, what do you all think?

grokys commented:
LGTM!
